.github/workflows/weekly.yml

```diff
@@ -13,7 +13,6 @@ jobs:
   latest-python:
```

Comment:
Hi @ana I'm not sure if I understood the problems that this is solving.

Could you please explain more on the commit description?

Reply:
> Hi @ana I'm not sure if I understood the problems that this is solving.
> 
> Could you please explain more on the commit description?

Hi @beraldoleal 
I've updated the commit message, hopefully it's more clear now!

Reply:
Thanks @ana for explaining it.

After reading the description, what about instead of duplicating the pipelines if we set properly the artifacts so we can download log messages to better debug?

Reply:
> Thanks @ana for explaining it.
> 
> After reading the description, what about instead of duplicating the pipelines if we set properly the artifacts so we can download log messages to better debug?

Mostly 2 reasons:
- I want to know quickly without debugging if this something that requires quick attention (problem in core avocado) or not (problem in a plugin) 
- Looking at the log messages you don't realize immediately they're caused by a plugin, you need to run the test locally.

Reply:
About 1: IMO, we should pay attention not only on "core" issues. Due the Avocado architecture, some  plugins are essential. Sometimes, a problem with a specific plugin might need a "quick attention" as well.

About 2: I agree that debugging in Avocado is a area for improvements, but IIUC, error messages are giving us the right hints (see bellow):

```
Failed tests:
check-892-selftests/functional/test_streams.py:StreamsTest.test_custom_stream_and_level FAIL
check-1033-optional_plugins/varianter_yaml_to_mux/tests/test_unit.py:TestMultipleLoaders.test_multiple_loaders FAIL
```

In some cases, yes, it is part of the debug process to configure a local environment so we can reproduce the problem. 
But, if we had the artifacts, we could look at the stdout/stderrors on each of those tests without the need of running locally on a specific environment.

Since we don't have that yet, I did a local test and seems to me that:

892: the problem here is that robot framework is not ready for the python3.10 syntax.
1033: yaml.safe_loader() for some reason is returning a different dict type in some cases (collections.OrderedDict). We are already discussing possible solutions on #4891 

If you think that is valuable to split the runs here, I will not be against it, but seems to me that the logs are showing exactly the what are our problems. And the artifacts could help a lot here. IMO, even more than the split.

Reply:
> About 1: IMO, we should pay attention not only on "core" issues. Due the Avocado architecture, some plugins are essential. Sometimes, a problem with a specific plugin might need a "quick attention" as well.

ACK

> About 2: I agree that debugging in Avocado is a area for improvements, but IIUC, error messages are giving us the right hints (see bellow):
> 
> ```
> Failed tests:
> check-892-selftests/functional/test_streams.py:StreamsTest.test_custom_stream_and_level FAIL
> check-1033-optional_plugins/varianter_yaml_to_mux/tests/test_unit.py:TestMultipleLoaders.test_multiple_loaders FAIL
> ```
> 
> In some cases, yes, it is part of the debug process to configure a local environment so we can reproduce the problem.
> But, if we had the artifacts, we could look at the stdout/stderrors on each of those tests without the need of running locally on a specific environment.
> 
> Since we don't have that yet, I did a local test and seems to me that:
> 
> 892: the problem here is that robot framework is not ready for the python3.10 syntax.

Yes, this was reported at https://github.com/avocado-framework/avocado/issues/4739
However, when running `python selftests/check.py --disable-plugin-checks robot` like at https://github.com/avocado-framework/avocado/runs/3396962071  The issue is still there, so there is some any other problem (and it's very likely with another plugin)

> 1033: yaml.safe_loader() for some reason is returning a different dict type in some cases (collections.OrderedDict). We are already discussing possible solutions on #4891
> 
> If you think that is valuable to split the runs here, I will not be against it, but seems to me that the logs are showing exactly the what are our problems. And the artifacts could help a lot here. IMO, even more than the split.

I'll add keeping the artifacts. One week is enough?

Reply:
> Yes, this was reported at #4739
> However, when running `python selftests/check.py --disable-plugin-checks robot` like at https://github.com/avocado-framework/avocado/runs/3396962071 The issue is still there, so there is some any other problem (and it's very likely with another plugin)

`--disable-plugin-checks robot` is disabling the plugin completely or just not running those tests? Even that we are not checking it, the plugin is still loaded: try to run `avocado run /bin/true` on python3.10. You will the error is still there.

The test is not inside the plugin, the test is `selftests/functional/test_streams.py:StreamsTest.test_custom_stream_and_level` and it is still capturing the problem. So I think that the test worked just fine here capturing something that is not supposed to be there.

> I'll add keeping the artifacts. One week is enough?

Since we are going to have the manual check, I think so.

Reply:
> > Yes, this was reported at #4739
> > However, when running `python selftests/check.py --disable-plugin-checks robot` like at https://github.com/avocado-framework/avocado/runs/3396962071 The issue is still there, so there is some any other problem (and it's very likely with another plugin)
> 
> Is `--disable-plugin-checks robot` disabling the plugin completely or just not running those tests? Even that we are not checking it, the plugin is still loaded: try to run `avocado run /bin/true` on python3.10. You will see that the error is still there.
> 
> The test is not inside the plugin, the test is `selftests/functional/test_streams.py:StreamsTest.test_custom_stream_and_level` and it is still capturing the problem. So I think that the test worked just fine here capturing something that is not supposed to be there.

No, if you don't install the plugins with `python3 setup.py develop --user --skip-optional-plugins` (the job in this PR does that)  you don't get any issue with `selftests/functional/test_streams.py:StreamsTest.test_custom_stream_and_level`

I didn't make any remark before about the installation without plugins and I should have insisted on this. Please, see the comment about updating this PR below for more information.